### ISSUE TITLE
Signedness Checker emits false positive with compound assignment operators

### ISSUE THREAD

**Issue by DmitriyShepelev** [typetools/checker-framework#3709]
The Signedness Checker emits a false positive with compound assignment operators:
```
public class CompoundAssignmentOperators {
    void additionWithCompoundAssignment(char c, int i1) {
        i1 += c;
    }

    void additionWithoutCompoundAssignment(char c, int i1) {
        i1 = (int) (i1 + c);
    }

    void subtractionWithCompoundAssignment(char c, int i1) {
        i1 -= c;
    }

    void subtractionWithoutCompoundAssignment(char c, int i1) {
        i1 = (int) (i1 - c);
    }

    void multiplicationWithCompoundAssignment(char c, int i1) {
        i1 *= c;
    }

    void multiplicationWithoutCompoundAssignment(char c, int i1) {
        i1 = (int) (i1 * c);
    }

    void divisionWithCompoundAssignment(char c, int i1) {
        i1 /= c;
    }

    void divisionWithoutCompoundAssignment(char c, int i1) {
        i1 = (int) (i1 / c);
    }

    void modulusWithCompoundAssignment(char c, int i1) {
        i1 %= c;
    }

    void modulusWithoutCompoundAssignment(char c, int i1) {
        i1 = (int) (i1 % c);
    }
}
```
Output:
```
root@DESKTOP-MO4BCEL:~/checker-framework# javacheck -processor SignednessChecker ~/temp-checker/checker-framework/checker/tests/signedness/Compoun
dAssignmentOperators.java
/root/temp-checker/checker-framework/checker/tests/signedness/CompoundAssignmentOperators.java:3: error: [compound.assignment.mixed.unsigned.expression] PLUS operation has signed LHS and unsigned RHS
        i1 += c;
              ^
  lhs: @Signed int
  rhs: @Unsigned char
/root/temp-checker/checker-framework/checker/tests/signedness/CompoundAssignmentOperators.java:11: error: [compound.assignment.mixed.unsigned.expression] MINUS operation has signed LHS and unsigned RHS
        i1 -= c;
              ^
  lhs: @Signed int
  rhs: @Unsigned char
/root/temp-checker/checker-framework/checker/tests/signedness/CompoundAssignmentOperators.java:19: error: [compound.assignment.mixed.unsigned.expression] MULTIPLY operation has signed LHS and unsigned RHS
        i1 *= c;
              ^
  lhs: @Signed int
  rhs: @Unsigned char
/root/temp-checker/checker-framework/checker/tests/signedness/CompoundAssignmentOperators.java:27: error: [compound.assignment.unsigned.expression] DIVIDE has unsigned RHS
        i1 /= c;
              ^
  lhs: @Signed int
  rhs: @Unsigned char
/root/temp-checker/checker-framework/checker/tests/signedness/CompoundAssignmentOperators.java:35: error: [compound.assignment.unsigned.expression] REMAINDER has unsigned RHS
        i1 %= c;
              ^
  lhs: @Signed int
  rhs: @Unsigned char
5 errors
```
`javacheck` alias:
`alias javacheck='$CHECKERFRAMEWORK/checker/bin/javac'`

According to [§15.26.2 Compound Assignment Operators](https://docs.oracle.com/javase/specs/jls/se14/html/jls-15.html#jls-15.26.2), `i1 += c;` in the method `additionWithCompoundOperator(int i1, char c)` is equivalent to `i1 = (int) (i1 + c);` in the method `additionWithoutCompoundOperator(int i1, char c)`. However, looking at the output, an error is emitted in the method `additionWithCompoundOperator(int i1, char c)` and **not** in the method `additionWithoutCompoundOperator(int i1, char c)`.
